src/Interpreters/BackupsStorage.cpp

```diff
@@ -0,0 +1,48 @@
+#include <Interpreters/BackupsStorage.h>
+
+#include <Interpreters/Context.h>
+#include <Interpreters/InterpreterAlterQuery.h>
+#include <Parsers/ParserAlterQuery.h>
+#include <Parsers/parseQuery.h>
+
+namespace DB
+{
+
+BackupsStorage::BackupsStorage(ContextPtr context_, const String & database, const String & table, const String & engine)
+    : SystemLogStorage(context_, database, table, engine)
+{
+    prepareTable();
+}
+
+void BackupsStorage::update(const BackupOperationInfo & info)
+{
+    /// Update current operation's entry
+    String query =
+        "ALTER TABLE " + table_id.getFullTableName() + " UPDATE"
+        + " `status` = " + std::to_string(static_cast<Int8>(info.status))
+        + ", `error` = " + quoteString(info.error_message)
+        + ", `end_time` = " + std::to_string(std::chrono::system_clock::to_time_t(info.end_time))
+        + ", `num_files` = " + std::to_string(info.num_files)
+        + ", `total_size` = " + std::to_string(info.total_size)
+        + ", `num_entries` = " + std::to_string(info.num_entries)
+        + ", `uncompressed_size` = " + std::to_string(info.uncompressed_size)
+        + ", `compressed_size` = " + std::to_string(info.compressed_size)
+        + ", `files_read` = " + std::to_string(info.num_read_files)
+        + ", `bytes_read` = " + std::to_string(info.num_read_bytes)
+        + " WHERE id = '" + info.id + "'";
```

Comment:
Interesting, implementing `UPDATE` with mutations...
Have you considered to use `ReplacingMergeTree` + adding new record in case of `update`?

Reply:
The original idea was to make it like `query_log` where we may specify any engine. For backups - any engine without replication (any engine which is not Replicated* and not KeeperMap). Finally in this PR we decided to use MergeTree for persistence.

Reply:
I'm just worrying about tons of mutations for backup information updates, yes the table is tiny, and maybe it is not a problem.

Though with the suggested alternative you can also allow `Memory` engine, but you just need to execute some query with aggregations applied, and maybe the whole `system.backups` should be rethinked in this way, like `query_log`, that you have multiple `status`'es and for each you can have different information.